### ISSUE TITLE
1.2.7版本，出现很多native层崩溃

### ISSUE THREAD

**Issue by mutour** [Tencent/MMKV#658]
最近项目多出很多崩溃，麻烦看下是啥原因。

使用版本：'com.tencent:mmkv-static:1.2.7'
全部采用的MULTI_PROCESS_MODE方式

```
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
        pid: 0, tid: 0 >>> com.xxx.xxx <<<

        backtrace:
        #00  pc 0000000000027b30  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/split_config.arm64_v8a.apk!lib/arm64-v8a/libmmkv.so (offset 0x172000)
        #00  pc 000000000001ba44  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/split_config.arm64_v8a.apk!lib/arm64-v8a/libmmkv.so (offset 0x172000)
        #00  pc 00000000000123e8  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/split_config.arm64_v8a.apk!lib/arm64-v8a/libmmkv.so (offset 0x172000)
        #00  pc 00000000001ed468  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (com.tencent.mmkv.MMKV.checkProcessMode [DEDUPED]+152)
        #00  pc 000000000041dd6c  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (com.tencent.mmkv.MMKV.à¸§+172)
        #00  pc 000000000041d990  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (com.tencent.mmkv.MMKV.à¸+112)
        #00  pc 00000000002bc738  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (à¸¯à¸à¸¤$à¸§.<init>+264)
        #00  pc 00000000002be5b8  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (à¸¯à¸à¸¤$à¸¨.<init>+264)
        #00  pc 0000000000281bd4  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (à¸¯à¸à¸¤.à¸+372)
        #00  pc 0000000000251054  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (com.xxxxxx.BaseApplication.getSharedPreferences+644)
        #00  pc 000000000039d1e4  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (com.flurry.sdk.fj.a+4916)
        #00  pc 000000000038d118  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (com.flurry.sdk.fp.a+120)
        #00  pc 000000000038d17c  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (com.flurry.sdk.fp.a+220)
        #00  pc 000000000038d17c  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (com.flurry.sdk.fp.a+220)
        #00  pc 000000000038d17c  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (com.flurry.sdk.fp.a+220)
        #00  pc 000000000038d17c  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (com.flurry.sdk.fp.a+220)
        #00  pc 000000000034ef30  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (com.flurry.sdk.a$1.a+160)
        #00  pc 0000000000317e1c  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (com.flurry.sdk.dy.run+60)
        #00  pc 00000000002ca384  /system/framework/arm64/boot-core-oj.oat (offset 0x114000) (java.util.concurrent.Executors$RunnableAdapter.call+68)
        #00  pc 000000000037098c  /system/framework/arm64/boot-core-oj.oat (offset 0x114000) (java.util.concurrent.FutureTask.run+204)
        #00  pc 000000000031ce4c  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (com.flurry.sdk.h$a.run+284)
        #00  pc 000000000037969c  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (com.flurry.sdk.f.c+396)
        #00  pc 000000000031cd94  /data/app/com.xxx.xxx-7Ifsdi1IhfM3JpyYmOuYAA==/oat/arm64/base.odex (offset 0x1ed000) (com.flurry.sdk.h$a.run+100)
        #00  pc 00000000003d9754  /system/framework/arm64/boot-core-oj.oat (offset 0x114000) (java.util.concurrent.ThreadPoolExecutor.runWorker+996)
        #00  pc 00000000003d64c0  /system/framework/arm64/boot-core-oj.oat (offset 0x114000) (java.util.concurrent.ThreadPoolExecutor$Worker.run+64)
        #00  pc 000000000025d128  /system/framework/arm64/boot-core-oj.oat (offset 0x114000) (java.lang.Thread.run+72)
        #00  pc 0000000000556188  /system/lib64/libart.so (art_quick_invoke_stub+584)
        #00  pc 00000000000cf8c8  /system/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+200)
        #00  pc 000000000045d04c  /system/lib64/libart.so (art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+104)
        #00  pc 000000000045e108  /system/lib64/libart.so (art::InvokeVirtualOrInterfaceWithJValues(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, jvalue*)+424)
        #00  pc 000000000048944c  /system/lib64/libart.so (art::Thread::CreateCallback(void*)+1120)
        #00  pc 00000000000922b8  /system/lib64/libc.so (__pthread_start(void*)+36)
        #00  pc 0000000000023b00  /system/lib64/libc.so (__start_thread+68)






        *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
        pid: 0, tid: 0 >>> com.xxx.xxx <<<

        backtrace:
        #00  pc 0000000000027b30  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/split_config.arm64_v8a.apk!lib/arm64-v8a/libmmkv.so (offset 0x172000)
        #00  pc 000000000001ba44  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/split_config.arm64_v8a.apk!lib/arm64-v8a/libmmkv.so (offset 0x172000)
        #00  pc 00000000000123e8  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/split_config.arm64_v8a.apk!lib/arm64-v8a/libmmkv.so (offset 0x172000)
        #00  pc 0000000000177428  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (art_jni_trampoline+152)
        #00  pc 000000000035e410  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (com.tencent.mmkv.MMKV.à¸§+176)
        #00  pc 000000000035e0a0  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (com.tencent.mmkv.MMKV.à¸+112)
        #00  pc 000000000022f69c  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (à¸¯à¸à¸¤$à¸§.<init>+268)
        #00  pc 0000000000231364  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (à¸¯à¸à¸¤$à¸¨.<init>+276)
        #00  pc 00000000001fba20  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (à¸¯à¸à¸¤.à¸+384)
        #00  pc 00000000001cb3b8  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (com.xxxxxxxxx.BaseApplication.getSharedPreferences+664)
        #00  pc 00000000002e1b5c  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (com.flurry.sdk.fj.a+4972)
        #00  pc 00000000002d4614  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (com.flurry.sdk.fp.a+116)
        #00  pc 00000000002d4678  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (com.flurry.sdk.fp.a+216)
        #00  pc 00000000002d4678  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (com.flurry.sdk.fp.a+216)
        #00  pc 00000000002d4678  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (com.flurry.sdk.fp.a+216)
        #00  pc 00000000002d4678  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (com.flurry.sdk.fp.a+216)
        #00  pc 00000000002a1970  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (com.flurry.sdk.a$1.a+160)
        #00  pc 00000000002703dc  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (com.flurry.sdk.dy.run+60)
        #00  pc 00000000001b3dc4  /apex/com.android.art/javalib/arm64/boot.oat (java.util.concurrent.Executors$RunnableAdapter.call+68)
        #00  pc 000000000023e3c8  /apex/com.android.art/javalib/arm64/boot.oat (java.util.concurrent.FutureTask.run+200)
        #00  pc 0000000000274ac0  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (com.flurry.sdk.h$a.run+288)
        #00  pc 00000000002c4598  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (com.flurry.sdk.f.c+408)
        #00  pc 0000000000274a04  /data/app/~~fNMuSzubrVgk89saXzOHhA==/com.xxx.xxx-nbuQnT8aoOhpFIQACrd5VQ==/oat/arm64/base.odex (com.flurry.sdk.h$a.run+100)
        #00  pc 000000000029d318  /apex/com.android.art/javalib/arm64/boot.oat (java.util.concurrent.ThreadPoolExecutor.runWorker+984)
        #00  pc 000000000029a510  /apex/com.android.art/javalib/arm64/boot.oat (java.util.concurrent.ThreadPoolExecutor$Worker.run+64)
        #00  pc 000000000015ab08  /apex/com.android.art/javalib/arm64/boot.oat (java.lang.Thread.run+72)
        #00  pc 0000000000133564  /apex/com.android.art/lib64/libart.so (art_quick_invoke_stub+548)
        #00  pc 00000000001a8a78  /apex/com.android.art/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+200)
        #00  pc 0000000000554c48  /apex/com.android.art/lib64/libart.so (art::JValue art::InvokeVirtualOrInterfaceWithJValues<art::ArtMethod*>(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, art::ArtMethod*, jvalue const*)+460)
        #00  pc 00000000005a3fe4  /apex/com.android.art/lib64/libart.so (art::Thread::CreateCallback(void*)+1308)
        #00  pc 00000000000afd4c  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+64)
        #00  pc 0000000000050288  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+64)
```

**Comment by lingol** [comment]
You should provide logs as well.

**Comment by lingol** [comment]
#630

**Comment by mutour** [comment]
> Assuming you are using arm64 abi, here are the symbolized call stacks:
> 
> > mmkv::FileLock::isFileLockValid()
> > /Users/lingol/Developer/mmkv/Core/InterProcessLock.h:48
> > mmkv::InterProcessLock::try_lock(bool*)
> > /Users/lingol/Developer/mmkv/Core/InterProcessLock.h:104
> > mmkv::checkProcessMode(_JNIEnv*, _jobject*, long)
> > /Users/lingol/Developer/mmkv/Android/MMKV/mmkv/.cxx/cmake/StaticCppRelease/arm64-v8a/../../../../src/main/cpp/native-bridge.cpp:349
> 
> You should provide logs as well.

该问题出现在google play console后台的崩溃信息，暂时无任何其他日志。

**Comment by lingol** [comment]
Without logs it's unlikely to further analyze this issue.

**Comment by lingol** [comment]
This issue will be closed this week if there's no further information provided.

**Comment by lingol** [comment]
Close due to inactive.

**Comment by lingol** [comment]
Released with [v1.2.8](https://github.com/Tencent/MMKV/releases/tag/v1.2.8).

**Comment by 532922397** [comment]
> Released with [v1.2.8](https://github.com/Tencent/MMKV/releases/tag/v1.2.8).

nice
